src/ui/public/accessibility/kbn_ui_ace_keyboard_mode.js

```diff
@@ -0,0 +1,65 @@
+/**
+ * The `kbn-ui-ace-keyboard-mode` directive should be used on any element, that
+ * `ui-ace` is used on. It will prevent the keyboard trap, that ui-ace usually
+ * has, i.e. tabbing into the box won't give you any possibilities to leave
+ * it via keyboard again, since tab inside the textbox works like a tab character.
+ *
+ * This directive won't change anything, if the user uses the mouse. But if she
+ * tabs to the ace editor, an overlay will be shown, that you have to press Enter
+ * to enter editing mode, and that it can be left by pressing Escape again.
+ *
+ * That way the ui-ace editor won't trap keyboard focus, and won't cause that
+ * accessibility issue anymore.
+ */
+
+import angular from 'angular';
+import { uiModules } from 'ui/modules';
+import './kbn_ui_ace_keyboard_mode.less';
+import { ESC_KEY_CODE, ENTER_KEY } from 'ui_framework/services';
+
+uiModules.get('kibana').directive('kbnUiAceKeyboardMode', () => ({
+  restrict: 'A',
+  link(scope, element) {
+    const uniqueId = `uiAceKeyboardHint-${scope.$id}-${Math.floor(Math.random() * 10000)}`;
```

Comment:
I used the same approach we used in several other places (date picker, typeahead). This brings us back to the topic, that we might want to create a universal HTML Id generation service, that we have a distinct way to create those ids :-)

But I will change it to your suggestion.